python/ray/serve/_private/deployment_state.py

```diff
@@ -1608,6 +1609,27 @@ def get_total_num_requests(self) -> float:
             RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE
             or len(running_replicas) == 0
         ):
+            # Drop metrics for handles that haven't sent an update in a while.
+            # This is expected behavior for handles that were on replicas or
+            # proxies that have been shut down.
+            timeout_s = max(
+                2 * self.autoscaling_policy_manager.get_metrics_interval_s(),
+                RAY_SERVE_MIN_HANDLE_METRICS_TIMEOUT_S,
+            )
+            for handle_id, handle_metric in list(self.handle_requests.items()):
+                if time.time() - handle_metric.timestamp >= timeout_s:
+                    del self.handle_requests[handle_id]
+                    total_ongoing_requests = handle_metric.queued_requests + sum(
+                        handle_metric.running_requests.values()
+                    )
+                    if total_ongoing_requests > 0:
+                        logger.info(
+                            f"Dropping stale metrics for handle '{handle_id}' "
+                            f"because no update was received for {timeout_s:.1f}s. "
+                            f"Ongoing requests was: "
+                            "{total_ongoing_requests}."
```

Comment:
```suggestion
                            f"Ongoing requests was: {total_ongoing_requests}."
```

Reply:
what the heck the linter sabotaged me here!